installer/scripts/tue-install-impl.bash

```diff
@@ -172,12 +172,37 @@ function tue-install-svn
 
 # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
 
+function _make-ssh-url-git
+{
+    local url=$1
+    # Reset global variable holding repo SSH url
+    git_repo_ssh_url=
+
+    # Check if url has https else exit
+    echo $url | grep -q "https://" || return 1
+
+    # Check if url is a git repository
+    echo $url | grep -q ".git" || return 1
+
+    local web_address=${url#https://}
+    local domain_name=${web_address%%/*}
+    local repo_address=${web_address#*/}
+
+    git_repo_ssh_url=git@$domain_name:$repo_address
+}
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+
 function tue-install-git
 {
+    # Generate SSH url
+    _make-ssh-url-git $1
+
     tue-install-debug "tue-install-git $@"
     if [ ! -d $2 ]; then
         tue-install-debug "git clone --recursive $1 $2"
-        res=$(git clone --recursive $1 $2 2>&1)
+        # If cloning with SSH fails use HTTPS
+        res=$(git clone --recursive $git_repo_ssh_url $2 2> /dev/null || git clone --recursive $1 $2 2>&1)
```

Comment:
I would like to see it as an IF which checks a parameter. Would be nice to add tooling for that, so add functions which can set the parameter.

I don't want SSH as the default.

Reply:
Parameter checking in what sense? 

And what's the harm in having SSH as default? It preferred over HTTPS because it is more secure.

Reply:
Few issues:
- What is the influence on speed of trying and failing if you are doing this 100x?
- Once you cloned this way, you can't switch between SSH and HTTPS, which is also limiting.
- In our tue-robocup-* functions we assume the use of HTTPS, see for example [_tue-add-git-remote](https://github.com/tue-robotics/tue-env/blob/master/setup/tue-functions.bash#L732) This would break by merging this.

Reply:
So I have added an IF condition but thinking about your first issue, do you want me to add a counter that after 5 consecutive failures will issue a warning and just switch to HTTPS?